pkg/webui/console/views/gateway-overview/gateway-overview.js

```diff
@@ -33,16 +39,44 @@ import GatewayConnection from '@console/containers/gateway-connection'
 
 import withFeatureRequirement from '@console/lib/components/with-feature-requirement'
 
+import downloadObjectAsFile from '@ttn-lw/lib/download-object-as-file'
+import { selectGsConfig } from '@ttn-lw/lib/selectors/env'
 import PropTypes from '@ttn-lw/lib/prop-types'
 import sharedMessages from '@ttn-lw/lib/shared-messages'
 
+import token from '@console/lib/access-token'
 import { mayViewGatewayInfo } from '@console/lib/feature-checks'
 
+const m = defineMessages({
+  downloadGlobalConf: 'Download global_conf.json',
+  showGlobalConf: 'Show global_conf.json',
+  globalConf: 'Global configuration',
+})
+
+async function getGlobalConf(gtwId, baseURL, tkn) {
+  return await api.gateway.getGlobalConf(gtwId, baseURL, tkn)
+}
+
 @withRequest(({ gtwId, loadData }) => loadData(gtwId), () => false)
 @withFeatureRequirement(mayViewGatewayInfo, {
   redirect: '/',
 })
 export default class GatewayOverview extends React.Component {
+  constructor(props) {
+    super(props)
+
+    this.state = {
+      globalConf: null,
+    }
+
+    const baseURL = selectGsConfig().base_url
+    const tkn = token()
+    const { gtwId } = props
+    this._asyncRequest = getGlobalConf(gtwId, baseURL, tkn).then(globalConf =>
+      this.setState({ globalConf }),
+    )
+  }
```

Comment:
1.  The token was already passed to the SDK, you should source it from within the SDK.
2.  It's an anti-pattern to fetch data in the component constructor, since you might get a response before the component is mounted (and `this.setState()` is effective). Use `componentDidMount()` lifecycle method instead.
3.  We use async / await for asynchronous logic
4.  It's also not necessary to fetch the data on component mount, since it is being displayed on demand, so you can launch the API request then and avoid unnecessary XHRs.